app/scripts/directives/oscWebhookTriggers.js

```diff
@@ -0,0 +1,153 @@
+"use strict";
+
+angular.module("openshiftConsole")
+
+  .directive("oscWebhookTriggers", function(
+    $uibModal,
+    $filter,
+    APIService) {
+
+    return {
+      restrict: 'E',
+      scope: {
+        webhookSecrets: "=",
+        namespace: "=",
+        type: "@",
+        webhookTriggers: "=",
+        form: '='
+      },
+      templateUrl: 'views/directives/osc-webhook-triggers.html',
+      controller: function($scope) {
+        $scope.secretsVersion = APIService.getPreferredVersion('secrets');
+        $scope.webhookTypesOptions = [{
+          type: 'github',
+          label: 'GitHub'
+        }, {
+          type: 'gitlab',
+          label: 'GitLab'
+        }, {
+          type: 'bitbucket',
+          label: 'Bitbucket'
+        }, {
+          type: 'generic',
+          label: 'Generic'
+        }];
+
+        // Check if the webhook trigger contains only deprecated secret format.
+        $scope.isDeprecated = function(trigger) {
+          var triggerSecretData = $filter('getWebhookSecretData')(trigger);
+          if (_.has(triggerSecretData, 'secret') && !_.has(triggerSecretData, 'secretReference.name')) {
+            return true;
+          }
+          return false;
+        };
+
+        $scope.togglePassword = function(index) {
+          var deprecatedInputField = $('.add-webhook-row')[index].find('.deprecated-secret input');
+          $(deprecatedInputField).attr('type') === 'password' ? $(deprecatedInputField).attr('type', 'text') : $(deprecatedInputField).attr('type', 'password');
+        };
+
+        $scope.removeWebhookTrigger = function(index) {
+          if ($scope.webhookTriggers.length === 1) {
+            var lonelyTrigger = _.first($scope.webhookTriggers);
+            lonelyTrigger.lastTriggerType = "";
+            lonelyTrigger.data = {
+              type: ""
+            };
+          } else {
+            $scope.webhookTriggers.splice(index,1);
+          }
+          $scope.form.$setDirty();
+
+          // After removing trigger check check other triggers for duplicates.
+          _.each($scope.webhookTriggers, function(trigger) {
+            if (trigger.isDuplicit) {
+              duplicityCheck(trigger);
+            }
+          })
+        };
+
+        // Check if new or modified webhook trigger is a duplicate. If so, show a warning under the appropriate trigger
+        var duplicityCheck = function(trigger) {
+          var matchingTriggers = _.filter($scope.webhookTriggers, function(wehbookTrigger) {
+            return _.isEqual(wehbookTrigger.data, trigger.data);
```

Comment:
Does `data` include the secret? Do we expect to have duplicates often?

I'm wondering if we really need to be checking duplicates. @jwforres opinion?

Reply:
yes it does.
After speaking with @jwforres and @ncameronbritt we decided to notify user about the dups.